### PR TITLE
Add type hints to some tests files

### PR DESCRIPTION
**Pull request by dklimpel**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: tests/storage/test_monthly_active_users.py [hunk 1]
```diff
@@ -24,15 +28,15 @@
 FORTY_DAYS = 40 * 24 * 60 * 60
 
 
-def gen_3pids(count):
+def gen_3pids(count: int):
```
Comment (DMRobertson) [inline tests/storage/test_monthly_active_users.py:31]:
```suggestion
def gen_3pids(count: int) -> Dict[str, Any]:
```

File: tests/test_phone_home.py [hunk 2]
```diff
@@ -16,23 +16,24 @@
...
+    def test_performance_frozen_clock(self) -> None:
         """
         If time doesn't move, don't error out.
         """
         past_stats = [
             (self.hs.get_clock().time(), resource.getrusage(resource.RUSAGE_SELF))
         ]
-        stats = {}
+        stats: JsonDict = {}
         self.get_success(phone_stats_home(self.hs, stats, past_stats))
```
Comment (DMRobertson) [inline tests/test_phone_home.py:33]: Aside: I wonder why this takes `stats` as an out parameter. Seems to go back to #6318.

File: mypy.ini [hunk 3]
```diff
@@ -76,7 +76,6 @@ exclude = (?x)
    |tests/storage/test_base.py
    |tests/storage/test_roommember.py
    |tests/test_metrics.py
-   |tests/test_phone_home.py
    |tests/test_server.py
    |tests/test_state.py
    |tests/test_terms_auth.py
```
Comment (DMRobertson) [inline mypy.ini:81]: I'm guessing the other three test files this change need additional work to pass mypy?

**Review by DMRobertson — CHANGES_REQUESTED** [review]
LGTM---just one minor thing

File: mypy.ini (hunk 3 above)
Comment (dklimpel) [reply mypy.ini:81]: `tests/storage/test_monthly_active_users.py` and `tests/storage/test__base.py` are not in exclusion list and is different to `tests/storage/test_base.py`. There is a `_` more.

There a more errors:
```
tests/storage/test_roommember.py:37: error: Argument 3 of "prepare" is incompatible with supertype "HomeserverTestCase"; supertype defines the argument type as "HomeServer"  [override]
```
I am do not understand why here is used `HomeserverTestCase`.

Comment (DMRobertson) [reply mypy.ini:81]: Thanks for explaining that. We can deal with those problems further down the line.

File: tests/storage/test_monthly_active_users.py [hunk 4]
```diff
@@ -28,7 +28,7 @@
 FORTY_DAYS = 40 * 24 * 60 * 60
 
 
-def gen_3pids(count: int):
+def gen_3pids(count: int) -> Dict[str, Any]:
```
Comment (DMRobertson) [inline tests/storage/test_monthly_active_users.py:31]: Oh sorry Dirk, I got the return type wrong here.

```suggestion
def gen_3pids(count: int) -> List[Dict[str, Any]]:
```

**Comment by DMRobertson** [comment]
Argh, I missed the import. Sorry again.

**Review by DMRobertson — APPROVED** [review]
Thanks for your continued efforts here Dirk!

**Pull request merged**